src/SME.SGP.Aplicacao/CasosDeUso/DiarioBordo/InserirDiarioBordoUseCase.cs

```diff
@@ -27,17 +27,31 @@ public async Task<AuditoriaDto> Executar(InserirDiarioBordoDto param)
 
             param.ComponenteCurricularId = await RetornaComponenteCurricularIdPrincipalDoProfessor(turma.CodigoTurma, param.ComponenteCurricularId);
 
+            if (param.ComponenteCurricularId == 0)
+                throw new NegocioException($"Componente Curricular não encontrado");
+
             var auditoria = await mediator.Send(new InserirDiarioBordoCommand(param.AulaId, param.Planejamento, param.ComponenteCurricularId));
             await mediator.Send(new ExcluirPendenciaDiarioBordoPorIdEComponenteIdCommand(param.AulaId, param.ComponenteCurricularId));
             return auditoria;
         }
         private async Task<long> RetornaComponenteCurricularIdPrincipalDoProfessor(string turmaCodigo, long componenteCurricularId)
         {
             var disciplinas = await consultasDisciplina.ObterComponentesCurricularesPorProfessorETurma(turmaCodigo, false, false, false);
-            if (disciplinas.Count() > 1)
-                return disciplinas.Any() ? disciplinas.FirstOrDefault(b => b.CodigoComponenteCurricular == componenteCurricularId).CodigoComponenteCurricular : 0;
+            if (disciplinas != null && disciplinas.Any())
+            {
+                if (disciplinas.Count() > 1)
+                {
+                    var disciplina = disciplinas.Where(b => b.CodigoComponenteCurricular == componenteCurricularId);
+
+                    if (disciplina.Any())
+                        return disciplina.FirstOrDefault().CodigoComponenteCurricular;
```

Comment:
cara, aqui poderia ter um teste de nulo, pra evitar o erro de "object reference not set" na hr de pegar a propriedade código do componente curricular.
Aí com o teste de nulo talvez tenha q retornar o valor 0.